src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -0,0 +1,329 @@
+import LoadingIndicator from 'shared/components/loadingIndicator/LoadingIndicator';
+import {
+    getSampleViewUrlWithPathname,
+    getPatientViewUrlWithPathname,
+} from 'shared/api/urls';
+import { getAllClinicalDataByStudyViewFilter } from '../StudyViewUtils';
+import { StudyViewPageStore } from 'pages/studyView/StudyViewPageStore';
+import { isUrl, remoteData } from 'cbioportal-frontend-commons';
+import { makeObservable, observable, computed } from 'mobx';
+import { ResourceData, StudyViewFilter } from 'cbioportal-ts-api-client';
+
+export interface IFilesLinksTable {
+    store: StudyViewPageStore;
+}
+
+class FilesLinksTableComponent extends LazyMobXTable<{
+    [id: string]: string | number;
+}> {}
+
+const RECORD_LIMIT = 500;
+
+async function fetchFilesLinksData(
+    filters: StudyViewFilter,
+    sampleIdResourceData: { [sampleId: string]: ResourceData[] },
+    searchTerm: string | undefined,
+    sortAttributeId: string | undefined,
+    sortDirection: 'asc' | 'desc' | undefined,
+    recordLimit: number
+) {
+    const studyClinicalDataResponse = await getAllClinicalDataByStudyViewFilter(
+        filters,
+        searchTerm,
+        sortAttributeId,
+        sortDirection,
+        recordLimit,
+        0
+    );
+    const getResourceDataOfPatients = async () => {
+        const resourcesPerPatient = _(studyClinicalDataResponse.data)
+            .flatMap(clinicaldataItems => clinicaldataItems)
+            .uniqBy('patientId')
+            .map(resource =>
+                internalClient.getAllResourceDataOfPatientInStudyUsingGET({
+                    studyId: resource.studyId,
+                    patientId: resource.patientId,
+                    projection: 'DETAILED',
+                })
+            )
+            .flatten()
+            .value();
+
+        return Promise.all(resourcesPerPatient).then(resourcesPerPatient =>
+            _(resourcesPerPatient)
+                .flatMap()
+                .groupBy('patientId')
+                .value()
+        );
+    };
+
+    const resourcesForPatients = await getResourceDataOfPatients();
+    const buildItemsAndResources = (resourceData: {
+        [key: string]: ResourceData[];
+    }) => {
+        const resourcesPerPatient: { [key: string]: number } = {};
+        const items: { [attributeId: string]: string | number }[] = _(
+            resourceData
+        )
+            .flatMap(data =>
+                data.map(resource => ({
+                    studyId: resource.studyId,
+                    patientId: resource.patientId,
+                    sampleId: resource.sampleId,
+                    resourcesPerPatient: 0,
+                    typeOfResource: resource?.resourceDefinition?.displayName,
+                    description: resource?.resourceDefinition?.description,
+                    url: resource?.url,
+                }))
+            )
+            .value();
+
+        _(resourceData).forEach(data => {
```

Comment:
not that important but for the future, this is a reduce operation.   so, it would be slightly cleaner: 
```
const resourcesPerPatient = _.reduce(resourceData, ()=>..., {})
```
the idea is always to use direct assignment instead of  declaration and foreach.